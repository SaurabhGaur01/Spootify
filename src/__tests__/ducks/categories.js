import {
    setCategoriesData,
    clearCategoriesData,
    categoriesDataReducer,
} from '../../ducks/categories';

const mockObject = [
    {
        mockVariable: 'mockValue',
    }
];

describe('categories Duck Tests', () => {
    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setCategoriesData should match the snapshot' , () => {
            expect(setCategoriesData(mockObject)).toMatchSnapshot();
        });
        it('clearCategoriesData should match the snapshot' , () => {
            expect(clearCategoriesData()).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(categoriesDataReducer()).toEqual([]);
        });
        it('should return the passed state if called with no action' , () => {
            expect(categoriesDataReducer(mockObject)).toEqual(mockObject);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(categoriesDataReducer(mockObject, unknownAction)).toEqual(mockObject);
        });
        it('should set the value when called with setCategoriesData' , () => {
            expect(categoriesDataReducer('', setCategoriesData(mockObject))).toEqual(mockObject);
        });
        it('should clear the store when called with clearCategoriesData' , () => {
            expect(categoriesDataReducer(mockObject, clearCategoriesData())).toEqual([]);
        });
    });
});