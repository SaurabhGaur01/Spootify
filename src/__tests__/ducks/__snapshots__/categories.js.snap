// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`categories Duck Tests Action Generator clearCategoriesData should match the snapshot 1`] = `
Object {
  "type": "modules/spootify-app/CLEAR_CATEGORIES_DATA",
}
`;

exports[`categories Duck Tests Action Generator setCategoriesData should match the snapshot 1`] = `
Object {
  "data": Array [
    Object {
      "mockVariable": "mockValue",
    },
  ],
  "type": "modules/spootify-app/SET_CATEGORIES_DATA",
}
`;
